src/pages/CreateAssignment/CreateAssignment.js

```diff
@@ -61,6 +66,20 @@ function CreateAssignment({ firebase }) {
     setShowPhonics(false);
     setShowWriting(true);
   }
+  function validateAssignment() {
+    if (lessonType != "C" && (wordGroup == null || words.length < 4)) {
+      setInvalidMessage("Please include at least 4 words.");
+      setValidAssignment(false);
+    } else if (deploymentAccountIds < 1) {
+      setInvalidMessage("Please assign to at least one student.");
+      setValidAssignment(false);
+    } else if (date == null) {
+      setInvalidMessage("Please select a date on the calendar.");
+      setValidAssignment(false);
```

Comment:
there's a lot of repetition here with setting `validAssignment` to false for each case- you could even use the `invalidMessage` state as a boolean by setting it to either `null` (false) or a message (true) instead of having two state variables that are logically connected